raiden/network/proxies/channel_manager.py

```diff
@@ -96,14 +99,51 @@ def new_netting_channel(self, other_peer: Address, settle_timeout: int) -> Addre
         if local_address == other_peer:
             raise SamePeerAddress('The other peer must not have the same address as the client.')
 
-        transaction_hash = estimate_and_transact(
-            self.proxy,
-            'newChannel',
+        # Check to see if a channel with this partner already exists
+        existing_channel = self.proxy.call(
+            'getChannelWith',
             other_peer,
-            settle_timeout,
         )
 
-        self.client.poll(unhexlify(transaction_hash), timeout=self.poll_timeout)
+        if existing_channel:
+            raise DuplicatedChannelError("Channel with given partner address already exists")
+
+        # Prevent concurrent attempts to open a channel with the same token and
+        # partner address.
+        if other_peer not in self.open_channel_transactions:
```

Comment:
This code can be simplified a bit. Because these operations don't context switch:
 
- checking if a key is in a dictionary
- creating a Semaphore (or an AsyncResult)
- setting a key in a dictionary

For the thread that created the lock we know the check `if not open_channel_lock.locked():` will also be `True`, so that if can be removed.

If instead of a `Semaphore` we use an `AsyncResult`, we can wait on it and get a result, which allows the thread sending the transaction to tell the waiting threads what is the transaction hash. Because of this, the mapping `open_channel_transactions` doesn't need to have a tuple, only the `AsyncResult`.

Once the transaction polling is done, we know the check with `getChannelWith` will throw, so we don't need the `AsyncResult` anymore, this allows us to delete the key from the `open_channel_transactions` map and let python's refcount delete the `AsyncResult`